From b4adf7645ff6c14e19682b9ef17e15756c5b0b14 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 30 Aug 2010 13:04:25 +0000
Subject: [PATCH] Fix building on eglibc

Seems like recent eglibc requires that you include sys/stat.h
---
 src/archive.c  |    1 +
 src/database.c |    1 +
 src/io.c       |    1 +
 src/package.c  |    1 +
 4 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/archive.c b/src/archive.c
index f60cb1b..3ac8a64 100644
--- a/src/archive.c
+++ b/src/archive.c
@@ -20,6 +20,7 @@
 #include <unistd.h>
 #include <sysexits.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
 #include <limits.h>
 #include <stdint.h>
 
diff --git a/src/database.c b/src/database.c
index 5dcd9e0..925aa4a 100644
--- a/src/database.c
+++ b/src/database.c
@@ -20,6 +20,7 @@
 #include <signal.h>
 #include <fnmatch.h>
 #include <sys/file.h>
+#include <sys/stat.h>
 
 #include "apk_defines.h"
 #include "apk_package.h"
diff --git a/src/io.c b/src/io.c
index 2b0c892..5100d41 100644
--- a/src/io.c
+++ b/src/io.c
@@ -15,6 +15,7 @@
 #include <unistd.h>
 #include <malloc.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 #include <pwd.h>
 #include <grp.h>
 
diff --git a/src/package.c b/src/package.c
index e73814a..8f15e7a 100644
--- a/src/package.c
+++ b/src/package.c
@@ -19,6 +19,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
 
 #include <openssl/pem.h>
 
-- 
1.7.2.2

